From 23aa889c90f31870279dd2371ac8bf37d9a6a277 Mon Sep 17 00:00:00 2001
From: Steve Kondik <steve@cyngn.com>
Date: Wed, 4 Mar 2015 07:43:44 +0100
Subject: [PATCH 93/95] libmedia: Fix crash during mediaserver restart

 * AudioFlinger may have shut down, there's no reason to segfault.

Change-Id: I7625ae4f6d7eab96260936c2f077776d961b3d5a
---
 media/libmedia/AudioRecord.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/media/libmedia/AudioRecord.cpp b/media/libmedia/AudioRecord.cpp
index ca3832d..2dd2704 100644
--- a/media/libmedia/AudioRecord.cpp
+++ b/media/libmedia/AudioRecord.cpp
@@ -475,6 +475,12 @@ status_t AudioRecord::openRecord_l(size_t epoch)
 
     sp<IMemory> iMem;           // for cblk
     sp<IMemory> bufferMem;
+
+    if (audioFlinger == NULL) {
+        ALOGE("AudioFlinger was NULL!");
+        return NO_INIT;
+    }
+
     sp<IAudioRecord> record = audioFlinger->openRecord(input,
                                                        mSampleRate, mFormat,
                                                        mChannelMask,
-- 
2.5.0

